### PR TITLE
Issue 2083 pg get db, table, index size

### PR DESCRIPTION
**Pull request by jorgeuribe-aiven**
# What changed, and why it matters
It would be useful to have an Aiven doc for reference when service user ask about disk usage in PG. 

fixes https://github.com/aiven/devportal/issues/2083

### CODE REVIEW

**Comment by wojcik-dorota** [comment]
@jorgeuribe-aiven thank you for adding this content! I'll be happy to review it if you don't mind. I've noticed Cloudflare deployment didn't work for this PR and the preview URL is not available. Could you please try to rebase so that we check if the deployment works this time? Thanks again :-)

**Comment by wojcik-dorota** [comment]
Please add this article to
- https://github.com/aiven/devportal/blob/main/docs/products/postgresql/howto.rst 
- https://github.com/aiven/devportal/blob/main/docs/products/postgresql/howto/list-dba-tasks.rst
Thank you!

**Comment by wojcik-dorota** [comment]
Please introduce all code blocks using
```
.. code-block:: bash
```
instead ``::``
Use a relevant programming language ofc.

File: docs/products/postgresql/howto/pg-object-size.rst [hunk 1]
```diff
@@ -0,0 +1,98 @@
+Get database, table and index disk space usage
+===============================================
+
+PostgreSQL® offers different commands and functions to get the database and object disk space usage, which could cause some confusion as the result may vary depending on the function used. 
+This article will show some of the available commands and functions that can be used and the difference among them.  
+
+Get database size
+-----------------
+The database size can be retrieved using the following psql command ``\l+ [ pattern ]``::
```
Comment (wojcik-dorota) [inline docs/products/postgresql/howto/pg-object-size.rst:9]:
```
Retrieve the database size using either the ``\l+ [ pattern ]`` command `` or the the ``pg_database_size`` function.

Use the ``\l+ [ pattern ]`` command
'''''''''''''''''''''''''''''''''''
```

File: docs/products/postgresql/howto/pg-object-size.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    (6 rows)
+    
+    testdb2=> \l+ testdb2
+                                                    List of databases
+      Name   |  Owner   | Encoding |   Collate   |    Ctype    | Access privileges | Size  | Tablespace | Description 
+    ---------+----------+----------+-------------+-------------+-------------------+-------+------------+-------------
+     testdb2 | avnadmin | UTF8     | en_US.UTF-8 | en_US.UTF-8 |                   | 66 MB | pg_default | 
+    (1 row)
+
+An alternative to get the database size is to use the ``pg_database_size`` function::
```
Comment (wojcik-dorota) [inline docs/products/postgresql/howto/pg-object-size.rst:26]:
```
Use the ``pg_database_size`` function
'''''''''''''''''''''''''''''''''''''
```

File: docs/products/postgresql/howto/pg-object-size.rst [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+Individual index size can be retrieved using the command ``\di+ [ pattern ]``::
+
+    testdb2=> \di+ mytable1_indx1 
+                                                    List of relations
+       Schema    |      Name      | Type  |  Owner   |  Table   | Persistence | Access method |  Size  | Description 
+    -------------+----------------+-------+----------+----------+-------------+---------------+--------+-------------
+     test_schema | mytable1_indx1 | index | myowner  | mytable1 | permanent   | btree         | 912 kB | 
+    (1 row)
+
+The function ``pg_indexes_size`` can be used to query the size for **all the indexes** that belong to a table::
```
Comment (wojcik-dorota) [inline docs/products/postgresql/howto/pg-object-size.rst:73]:
```
Use the ``pg_indexes_size`` function
''''''''''''''''''''''''''''''''''''

The ``pg_indexes_size`` function can be used to query the size for all the indexes that belong to a table.
```

File: docs/products/postgresql/howto/pg-object-size.rst [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+    testdb2=> select pg_size_pretty(pg_indexes_size('mytable1')); 
+     pg_size_pretty 
+    ----------------
+     912 kB  
+    (1 row)  
+    
+Get table and its indexes size
```
Comment (wojcik-dorota) [inline docs/products/postgresql/howto/pg-object-size.rst:81]: Get a size for a table and its indices

File: docs/products/postgresql/howto/pg-object-size.rst [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+--------------------------------
+It is possible to get the disk usage for a table and its corresponding indexes, the function ``pg_total_relation_size`` computes the total disk space used by the specified table, including all indexes and TOAST data::
```
Comment (wojcik-dorota) [inline docs/products/postgresql/howto/pg-object-size.rst:83]: To get disk space usage for a table and its indexes, you can use the ``pg_total_relation_size`` function, which computes the total disk space used by the table, all its indices, and TOAST data.

File: docs/products/postgresql/howto/pg-object-size.rst [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    (1 row)     
+
+.. Warning:: 
+    It is not recommended to use the ``pg_relation_size`` function as it computes the disk space used by one “fork” of the specified relation. (Note that for most purposes it is more convenient to use the higher-level functions ``pg_total_relation_size`` or ``pg_table_size``, which sum the sizes of all forks.)
+.. Tip::
+    WAL files will also contribute to the service disk usage for additional information see: :doc:`../concepts/pg-disk-usage`.
+    
+    Relevant documentation is available on following links:  
+    `PostgreSQL interactive terminal <https://www.postgresql.org/docs/15/app-psql.html>`_ and
+    `Database Object Management Functions <https://www.postgresql.org/docs/current/functions-admin.html#FUNCTIONS-ADMIN-DBOBJECT>`_
```
Comment (wojcik-dorota) [inline docs/products/postgresql/howto/pg-object-size.rst:98]:
```
.. seealso::

   * `PostgreSQL interactive terminal <https://www.postgresql.org/docs/15/app-psql.html>`_
   * `Database Object Management Functions <https://www.postgresql.org/docs/current/functions-admin.html#FUNCTIONS-ADMIN-DBOBJECT>`_
```

File: docs/products/postgresql/howto/pg-object-size.rst [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+    testdb2=> select pg_size_pretty(pg_total_relation_size('mytable1')); 
+     pg_size_pretty 
+    ----------------
+     15 MB
+    (1 row)     
+
+.. Warning:: 
+    It is not recommended to use the ``pg_relation_size`` function as it computes the disk space used by one “fork” of the specified relation. (Note that for most purposes it is more convenient to use the higher-level functions ``pg_total_relation_size`` or ``pg_table_size``, which sum the sizes of all forks.)
+.. Tip::
+    WAL files will also contribute to the service disk usage for additional information see: :doc:`../concepts/pg-disk-usage`.
```
Comment (wojcik-dorota) [inline docs/products/postgresql/howto/pg-object-size.rst:94]:
```
WAL files also contribute to the service disk usage. For more information, check :doc:`About PostgreSQL® disk usage </docs/products/postgresql/concepts/pg-disk-usage>`.
```

File: docs/products/postgresql/howto/pg-object-size.rst [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+It is possible to get the disk usage for a table and its corresponding indexes, the function ``pg_total_relation_size`` computes the total disk space used by the specified table, including all indexes and TOAST data::
+
+    testdb2=> select pg_size_pretty(pg_total_relation_size('mytable1')); 
+     pg_size_pretty 
+    ----------------
+     15 MB
+    (1 row)     
+
+.. Warning:: 
+    It is not recommended to use the ``pg_relation_size`` function as it computes the disk space used by one “fork” of the specified relation. (Note that for most purposes it is more convenient to use the higher-level functions ``pg_total_relation_size`` or ``pg_table_size``, which sum the sizes of all forks.)
```
Comment (wojcik-dorota) [inline docs/products/postgresql/howto/pg-object-size.rst:92]:
```
It is not recommended to use the ``pg_relation_size`` function as it computes the disk space used by only one fork of the relation. To get the total size of all the relation's forks, use higher-level functions ``pg_total_relation_size`` or ``pg_table_size``.
```

**Review by wojcik-dorota — CHANGES_REQUESTED** [review]

**Comment by jorgeuribe-aiven** [comment]
@wojcik-dorota I have applied the changes you suggested and merged the main branch, so preview should be available.

**Comment by wojcik-dorota** [comment]
> @wojcik-dorota I have applied the changes you suggested and merged the main branch, so preview should be available.

@jorgeuribe-aiven looks great! I understand this has been a significant effort and thank you for that!
One last missing thing is: Please add this article to

- https://github.com/aiven/devportal/blob/main/docs/products/postgresql/howto.rst
- https://github.com/aiven/devportal/blob/main/docs/products/postgresql/howto/list-dba-tasks.rst

so that it's visible in relevant index/listing/ToC pages.

**Review by wojcik-dorota — APPROVED** [review]
LGTM! Thank you :-)

**Pull request merged**